src/status_im/hardwallet/core.cljs

```diff
@@ -1310,22 +1352,14 @@
 
 (fx/defn on-generate-and-load-key-success
   [{:keys [db random-guid-generator] :as cofx} data]
-  (let [{:keys [whisper-public-key
-                whisper-private-key
-                whisper-address
-                wallet-address
-                instance-uid
-                encryption-public-key]} (js->clj data :keywordize-keys true)
-        whisper-public-key' (str "0x" whisper-public-key)
-        instance-uid' (get-in db [:hardwallet :keycard-instance-uid])]
+  (let [account-data (js->clj data :keywordize-keys true)]
     (fx/merge cofx
               {:db (-> db
-                       (assoc-in [:hardwallet :whisper-public-key] whisper-public-key')
-                       (assoc-in [:hardwallet :whisper-private-key] whisper-private-key)
-                       (assoc-in [:hardwallet :whisper-address] whisper-address)
-                       (assoc-in [:hardwallet :wallet-address] wallet-address)
-                       (assoc-in [:hardwallet :encryption-public-key] encryption-public-key)
-                       (assoc-in [:hardwallet :keycard-instance-uid] (or instance-uid' instance-uid))
+                       (assoc-in [:hardwallet :account] (-> account-data
+                                                            (update :whisper-public-key #(str "0x" %))
+                                                            (update :instance-uid #(or (get-in db [:hardwallet :account :instance-uid])
```

Comment:
get-in already has an arity for default value so you can just write it
`(get-in db [:hardwallet :account :instance-uid] %)`